web3/providers/websocket/websocket_v2.py

```diff
@@ -67,7 +66,7 @@ def __init__(
         self,
         endpoint_uri: Optional[Union[URI, str]] = None,
         websocket_kwargs: Optional[Dict[str, Any]] = None,
-        call_timeout: Optional[int] = None,
+        call_timeout: Optional[float] = DEFAULT_PERSISTENT_CONNECTION_TIMEOUT,
```

Comment:
The timeout may be `None` or could be the max float value, `wait_for` will in turn wait forever (or until the timeout of the large value provided). Since this would be intentionally set by the user, I think it is acceptable to let the connection stay open until the response id is returned, if ever. What do you think?

Reply:
Hmm. That's a good point. We use this not only for call timeouts but for the while loop timeout. What if we set the while loop timeout to `self.call_timeout or 20` so that if it is set to `None` for call timeouts, the while loop will still max out in `20` seconds. Or whatever we decide that default is. Since it differs from the timeout for each call, I think that might be a good alternative.

Reply:
Makes sense to have the timeout limit on the while loop, perhaps we use the `call_timeout` unless it exceeds some threshold (perhaps even just 20s). That way it can short circuit as needed.

Reply:
My suggestion would've only worked for `None` but I think yours works well.

```py
timeout = self.call_timeout if self.call_timeout and self.call_timeout <= 20 else 20
```

This also prevents it from being `0` in the `if` check. Anything from `0.1` to `20` is fair game.